fastjsonschema/generator.py

```diff
@@ -367,29 +397,40 @@ def generate_items(self):
                             '{}[{{{}_x}}]'.format(self._variable_name, self._variable),
                         )
         else:
-            with self.l('for {variable}_x, {variable}_item in enumerate({variable}):'):
-                self.generate_func_code_block(
-                    self._definition['items'],
-                    '{}_item'.format(self._variable),
-                    '{}[{{{}_x}}]'.format(self._variable_name, self._variable),
-                )
+            if self._definition['items']:
+                with self.l('for {variable}_x, {variable}_item in enumerate({variable}):'):
+                    self.generate_func_code_block(
+                        self._definition['items'],
+                        '{}_item'.format(self._variable),
+                        '{}[{{{}_x}}]'.format(self._variable_name, self._variable),
+                    )
 
     def generate_min_properties(self):
+        with self.l('if not isinstance({variable}, dict):'):
+            self.l('return {variable}')
         self.create_variable_with_length()
         with self.l('if {variable}_len < {minProperties}:'):
             self.l('raise JsonSchemaException("{name} must contain at least {minProperties} properties")')
 
     def generate_max_properties(self):
+        with self.l('if not isinstance({variable}, dict):'):
+            self.l('return {variable}')
         self.create_variable_with_length()
         with self.l('if {variable}_len > {maxProperties}:'):
             self.l('raise JsonSchemaException("{name} must contain less than or equal to {maxProperties} properties")')
 
     def generate_required(self):
+        with self.l('if not isinstance({variable}, dict):'):
+            self.l('return {variable}')
         self.create_variable_with_length()
         with self.l('if not all(prop in {variable} for prop in {required}):'):
             self.l('raise JsonSchemaException("{name} must contain {required} properties")')
 
     def generate_properties(self):
+        with self.l('try:'):
+            self.l('{variable}.keys()')
+        with self.l('except AttributeError:'):
+            self.l('return {variable}')
```

Comment:
Wouldn't be simple condition faster than try-except block?

Reply:
That's likely.  I will refactor any more of these blocks to `isinstance` checks to align with the rest of the code.  My mistake.

Reply:
No worry, great, thanks!